pkg/planner/core/casetest/index/testdata/index_range_in.json

```diff
@@ -19,5 +19,31 @@
 	"select /*+ USE_INDEX(t1char,PKX) */ count(*) from t1char where  ((a1<'10') or (a1='10' and b1 < '20')) -- upper bound on (a1,b1)",
 	"select /*+ USE_INDEX(t1char,PKX) */ count(*) from t1char where ((a1>'1') or (a1='1' and b1 > '10')) and ((a1<'10') or (a1='10' and b1 < '20')) -- upper and lower bound on (a1,b1)"
     ]
+  },
+  {
+    "name": "TestRangeIntersection",
+    "cases": [
+	"select /*+ USE_INDEX(t1,PKX) */ count(*) from t1 where ((a1>1) or (a1=1 and b1 > 10));",
+	"select /*+ USE_INDEX(t1,PKX) */ count(*) from t1 where ((a1<10) or (a1=10 and b1 < 20));",
+	"select /*+ USE_INDEX(t1,PKX) */ count(*) from t1 where ((a1>1) or (a1=1 and b1 > 10)) and ((a1<10) or (a1=10 and b1 < 20));",
+	"select /*+ USE_INDEX(t1,PKX) */ count(*) from t1 where (a1,b1) > (1,10) and (a1,b1) < (10,20); -- row format of previous test",
+	"select /*+ USE_INDEX(t1,PKX) */ count(*) from t1 where ((a1>1) or (a1=1 and b1 >= 10)) and ((a1<10) or (a1=10 and b1 <= 20));",
+	"select /*+ USE_INDEX(t1,PKX) */ count(*) from t1 where (a1,b1) >= (1,10) and (a1,b1) <= (10,20); -- row format of previous test",
```

Comment:
Could you please add some no intersection range case ? Such as:
``` 
select count(*) from t1 where (a1, b1) > (1, 10) and (a1, b1) < (0, 20) 
select count(*) from t1 where (a1, b1) > (1, 10) and (a1, b1) < (2, 20) and b1 <5
select count(*) from t1 where (a1, b1) > (1, 10) and (a1, b1) < (4, 20) and a1 < 0
```

Reply:
The unit test for intersection covers empty intersections. But, added those as well.

Reply:
So, if the condition 'b1<5' doesn't begin as a1, it will not be merged with other condition range in our logic, right ?
But the actually range of "(a1, b1) > (1, 10) and (a1, b1) < (2, 20) and b1 <5"  should be empty.
So, we can't due with this situation right now ?

Reply:
Not really. The predicate is actually not empty. The row function is lexicographically ordered and (a1,b1) > (1,10) **does not** mean a1 > 1 and b1 > 10 which makes it with b1 < 5 empty.   (a1,b1) > (1,10) means (a1 > 1) or (a1=1 and b1 > 10) which does not contradict with b1 < 5. But, it does contradict with a1 < 0.